### PR TITLE
Raise an event when the page is really going to be unloaded (#667). Part 4 - event preventing.

### PR DESCRIPTION
**Pull request by churkin**
\cc @AlexanderMoskovkin @LavrovArtem

### CODE REVIEW

**Comment by churkin** [comment]
@testcafe-build-bot \retest

**Comment by churkin** [comment]
FPR

File: src/client/sandbox/event/listeners.js [hunk 1]
```diff
@@ -66,10 +69,21 @@ export default class Listeners extends EventEmitter {
             if (typeof eventCtx.outerHandlersWrapper === 'function')
                 return eventCtx.outerHandlersWrapper.call(this, e, listener);
 
+            var result = void 0;
+
             if (isObjectEventListener(listener))
-                return listener.handleEvent.call(listener, e);
+                result = listener.handleEvent.call(listener, e);
```
Comment (LavrovArtem) [inline src/client/sandbox/event/listeners.js]:
```js
result = listener.handleEvent(e);
```

**Comment by LavrovArtem** [comment]
\r-

**Comment by churkin** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm in general

I propose to add to the tests the check that `unload` event was not raised before the `redirectDetected` event. It's important for the `testcafe` repo. I can do this in this PR or in a following one

**Comment by churkin** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by LavrovArtem** [comment]
lgtm

**Pull request merged**